var/ramble/repos/builtin/modifiers/intel-aps/modifier.py

```diff
@@ -39,6 +55,33 @@ class IntelAps(BasicModifier):
         "mpi_command", " aps {aps_flags} ", method="append", modes=["mpi"]
     )
 
+    modifier_variable(
+        "aps_stat_level",
+        default="1",
+        description="Used to define the APS_STAT_LEVEL env variable",
+        mode="mpi",
+    )
+
+    modifier_variable(
+        "aps_extra_reports",
+        default="",
+        description="""
+        Comma-separated descriptors specifying extra reports (besides the summary) to generate.
+        Syntax definition:
+            aps_extra_reports = spec { "," spec }
+            spec = pre_defined_spec | custom_spec | "all"
+            pre_defined_spec = <keys in _PREDEFINED_REPORTS>
+            custom_spec = "custom" ":" options
+            options = <letters>
+        Examples:
+        * "transfer-graph,message-sizes,custom:-t"
+          Generates transfer comm. graph, message-size summary and MPI time per rank chart
+        * "all"
+          Generates all graphs defined in _PREDEFINED_REPORTS
+        """,
+        mode="mpi",
```

Comment:
Would this only work for `mode="mpi"`? Or should the `_PREDEFINED_REPORTS` dict have a mode sub-dict, to separate report types when we add new modes?

This might be more of a request to add a TODO, so we can update this later.

Reply:
Great call. Yes these reports only make sense for mpi mode. Changed to use a per-mode dict.